src/extension/flutter/flutter_daemon.ts

```diff
@@ -47,15 +48,20 @@ export class FlutterDaemon extends StdIOService<UnknownNotification> implements
 		if (process.env.DART_CODE_IS_TEST_RUN)
 			daemonArgs.push("--show-test-device");
 
-		const execution = usingCustomScript(
-			path.join(workspaceContext.sdks.flutter, flutterPath),
-			["daemon"].concat(daemonArgs),
-			workspaceContext.config?.flutterDaemonScript,
-		);
 
-		const flutterAdditionalArgs = config.for(vs.Uri.file(folder)).flutterAdditionalArgs;
-		const args = getGlobalFlutterArgs().concat(flutterAdditionalArgs).concat(execution.args);
-		this.createProcess(folder, execution.executable, args, { toolEnv: getToolEnv() });
+		if (!!config.daemonPort) {
+			this.createNcProcess(config.daemonPort);
+		} else {
+			const execution = usingCustomScript(
+				path.join(workspaceContext.sdks.flutter, flutterPath),
+				["daemon"].concat(daemonArgs),
+				workspaceContext.config?.flutterDaemonScript,
```

Comment:
Is this redundant now? Will internal uses always have both `flutterDaemonScript` set and `config.daemonPort` or is it valid to use just the former?

Reply:
Unfortunately this isn't redundant 😞 The `flutterDaemonScript` will be used when a user wants to access devices that are on the same machine as the extension host, using `config.daemonPort` will be for when the devices are on a different machine.

Reply:
Ah, gotcha :)